### PR TITLE
Updating multipart upload to use twine consume

### PR DESCRIPTION
**Pull request by nhibberd**

### CODE REVIEW

File: mismi-s3/test/Test/IO/Mismi/S3/Commands.hs [hunk 1]
```diff
@@ -156,8 +156,12 @@ prop_upload_fail d l = testAWS $ do
   liftIO . D.createDirectoryIfMissing True $ F.takeDirectory t
   liftIO $ T.writeFile t d
   uploadWithModeOrFail Fail t a
-  r <- uploadWithMode Fail t a
-  pure $ r === UploadError (UploadDestinationExists a)
+  r <- runEitherT $ uploadWithMode Fail t a
+  pure $ case r of
+    Left (UploadDestinationExists _) ->
```
Comment (charleso) [inline mismi-s3/test/Test/IO/Mismi/S3/Commands.hs]: Could you leave this as:

```haskell
r <- runEitherT (uploadWithMode Fail t a)
pure $ r === Left (UploadDestinationExists a)
```

Comment (nhibberd) [inline mismi-s3/test/Test/IO/Mismi/S3/Commands.hs]: No instance Eq on UploadError :(

**Comment by charleso** [comment]
:+1: What's the perf difference like?

File: mismi-s3/src/Mismi/S3/Commands.hs [hunk 2]
```diff
@@ -192,84 +187,104 @@ move source destination' =
...
+  let abort =
+        void . send . fencode' abortMultipartUpload a $ mpu
+
+  case r of
+    Left f ->
+      abort >>
+        (left $ MultipartUploadError f)
+
+    Right prts -> do
+      let z = sortOn (\(PartResponse i _) -> i) $ snd prts
```
Comment (charleso) [inline mismi-s3/src/Mismi/S3/Commands.hs]: As discussed maybe a comment on why the sort is required

**Pull request merged**